### ISSUE TITLE
Generate BOM entry in generated version catalog

### ISSUE THREAD

**Issue by cloudshiftchris** [austinarbor/version-catalog-generator#153]
Generating a BOM like below requires specifying the BOM version twice:

```
generate("jackson") {
            from("com.fasterxml.jackson:jackson-bom:2.18.0")
            aliasPrefixGenerator = GeneratorConfig.NO_PREFIX
            aliasSuffixGenerator = { _, _, artifactId ->
                artifactId.removePrefix("jackson-")
            }
        }
```

Requires an entry in libs.versions.toml:

`jackson-bom = { module = "com.fasterxml.jackson:jackson-bom", version = "2.18.0" }`

...to have this in the build script:

```
    implementation(platform(libs.jackson.bom))
    implementation(jackson.core)
    implementation(jackson.module.kotlin)
```

Ideally a "bom" entry (alias would be configurable) would be generated indicating the BOM the catalog came from.
This would eliminate the need for a BOM entry in libs.versions.toml and make the build script consistent in terms of library references:

```
    implementation(platform(jackson.bom))  // no "libs" prefix!
    implementation(jackson.core)
    implementation(jackson.module.kotlin)
```

**Comment by cloudshiftchris** [comment]
Am aware that you can use `from(toml("springBootDependencies"))` to avoid respecifying the version, but that still leaves a naming inconsistency between the BOM library reference and the other library references.

**Comment by austinarbor** [comment]
> that still leaves a naming inconsistency between the BOM library reference and the other library references.
do you mean that it would be `libs.jackson.bom` instead of `jackson.bom` ?

**Comment by cloudshiftchris** [comment]
Yes, the difference between
```
    implementation(platform(libs.jackson.bom))
    implementation(jackson.core)
    implementation(jackson.module.kotlin)
```

and

```
    implementation(platform(jackson.bom)) // generated
    implementation(jackson.core)
    implementation(jackson.module.kotlin)
````

**Comment by austinarbor** [comment]
@cloudshiftchris would you want to be able to use a custom alias for the bom entry, or is the existing logic using the actual artifact id ok?

**Comment by cloudshiftchris** [comment]
@austinarbor  the existing logic using the actual artifact id seems Ok, and preferable for consistency.

Perhaps there are some rare cases where a custom alias would be required, nothing seems obvious atm.

**Comment by austinarbor** [comment]
I have published 2.1.0 which includes this feature

```kotlin
generate("springLibs") {
      from(toml("springBootDependencies"))
      generateBomEntry = true
```

**Comment by cloudshiftchris** [comment]
Amazing.  Thank you.

Looking good.

This configuration:

```
        generate("jackson") {
            from("com.fasterxml.jackson:jackson-bom:2.18.0")
            aliasPrefixGenerator = GeneratorConfig.NO_PREFIX
            aliasSuffixGenerator = { _, _, artifactId -> artifactId.removePrefix("jackson-") }
            generateBomEntry = true
        }
```

Allows for consistent naming in dependencies blocks and removing the BOM entry from `libs.versions.toml`:
```
    implementation(platform(jackson.bom))
    implementation(jackson.core)
    implementation(jackson.module.kotlin)
    implementation(jackson.datatype.jsr310)
```
